src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ARRAY_SIZE = 10;
+    private int size = 0;
+    private int index;
+    private Object[] array = new Object[START_ARRAY_SIZE];
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            array = Arrays.copyOf(array, (int) (array.length * 1.5));
```

Comment:
1.5 should be GROW_FACTOR constant, for example

Reply:
Why Arrays.copyOf is not allowed. I reckon It's the best solution. Because if we here apply System.arraycopy, initially we should create new array with new size, then copy old to new one, then again copy to main array. It's too much extra code, don't you think ?

Reply:
I reckon that in some cases System.arraycopy is more logically, I have applied it when it reduce code, but in this case it add extra lines of code